pytorch_lightning/strategies/executors/ddp.py

```diff
@@ -0,0 +1,117 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import subprocess
+import sys
+from time import sleep
+from typing import Optional
+
+import __main__
+import numpy as np
+
+from pytorch_lightning.strategies.executors.base import Executor
+from pytorch_lightning.utilities import _HYDRA_AVAILABLE
+
+if _HYDRA_AVAILABLE:
+    from hydra.core.hydra_config import HydraConfig
+    from hydra.utils import get_original_cwd, to_absolute_path
+
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class DDPSubprocessExecutor(Executor):
+    def __init__(self, strategy):
+        super().__init__(strategy=strategy)
+        self.interactive_ddp_procs = []
+
+    def execute(self, trainer, function, *args, **kwargs):
+        self._call_children_scripts()
+        return function(*args, **kwargs)
+
+    def _call_children_scripts(self):
+        # bookkeeping of spawned processes
+        self._check_can_spawn_children()
+
+        # DDP Environment variables
+        os.environ["MASTER_ADDR"] = self.strategy.cluster_environment.main_address
```

Comment:
yes, but the issue here is that for DDP spawn, we need strategy since it's being used at a lot of places in the spawn function. Since we pass in the trainer to recover results, we can rely on the trainer argument to extract strategy, cluster environment, almost everything and remove any argument inside executor init?